src/devices/Vl53L0X/Vl53L0X.cs

```diff
@@ -0,0 +1,1163 @@
+            WriteRegister((byte)Registers.VHV_CONFIG_PAD_SCL_SDA__EXTSUP_HV, ReadByte((byte)(Registers.VHV_CONFIG_PAD_SCL_SDA__EXTSUP_HV) | 0x01));
+            WriteRegister(0x88, 0x00);
+            WriteRegister((byte)Registers.POWER_MANAGEMENT_GO1_POWER_FORCE, 0x01);
+            WriteRegister(0xFF, 0x01);
+            WriteRegister((byte)Registers.SYSRANGE_START, 0x00);
+            _stopData = ReadByte(0x91);
+            WriteRegister((byte)Registers.SYSRANGE_START, 0x01);
+            WriteRegister(0xFF, 0x00);
+            WriteRegister((byte)Registers.POWER_MANAGEMENT_GO1_POWER_FORCE, 0x00);
+
+            // disable SIGNAL_RATE_MSRC (bit 1) and SIGNAL_RATE_PRE_RANGE (bit 4) limit checks
+            WriteRegister((byte)Registers.MSRC_CONFIG_CONTROL, ReadByte((byte)(Registers.MSRC_CONFIG_CONTROL) | 0x12));
+            // set final range signal rate limit to 0.25 MCPS (million counts per second)
+            SetSignalRateLimit(0.25);
+
+            // Start intialization sequence
+            WriteRegister((byte)Registers.SYSTEM_SEQUENCE_CONFIG, 0xFF);
+            var spad = GetSpadInfo();
+
+            // Get the SPAD information
+            _i2cDevice.WriteByte((byte)Registers.GLOBAL_CONFIG_SPAD_ENABLES_REF_0);
+            // Allocate 1 more byte as it will be used to send back the configuration
+            Span<byte> ReferenceSpadMap = stackalloc byte[7];
+            // Skip the first byte for reading, it will be used for writting
+            _i2cDevice.Read(ReferenceSpadMap.Slice(1));
+
+            // Set the squads, prepare the registers firsts
+            WriteRegister(0xFF, 0x01);
+            WriteRegister((byte)Registers.DYNAMIC_SPAD_REF_EN_START_OFFSET, 0x00);
+            WriteRegister((byte)Registers.DYNAMIC_SPAD_NUM_REQUESTED_REF_SPAD, 0x2C);
+            WriteRegister(0xFF, 0x00);
+            WriteRegister((byte)Registers.GLOBAL_CONFIG_REF_EN_START_SELECT, 0xB4);
+
+            int firstSpadToEnable = spad.TypeIsAperture ? 12 : 0;
+            int squadEnable = 0;
+
+            for (int i = 0; i < 48; i++)
+            {
+                if ((i < firstSpadToEnable) || (squadEnable == spad.Count))
+                {
+                    // Enable only the SPAD that ware not yet
+                    ReferenceSpadMap[i / 8 + 1] &= (byte)~(1 << (i % 8));
+                }
+                else if ((ReferenceSpadMap[i / 8 + 1] >> ((i % 8)) & 0x1) == 0x01)
+                {
+                    squadEnable++;
+                }
+            }
+            // Write back the SPAD configuration
+            ReferenceSpadMap[0] = (byte)Registers.GLOBAL_CONFIG_SPAD_ENABLES_REF_0;
+            _i2cDevice.Write(ReferenceSpadMap);
+
+            // Those commands come from the offical API C code from STM
+            // Documentation does not provide register names
+            // In the main API, this block of command are hard coded
+            // The names of Registers provided are based on know ones.
```

Comment:
bummer but doesn't seem like we can do much about it without spending a lot of time trying to to dig where this info comes from

Reply:
I've already renamed some and for the rest, the original API is just a large array of bytes. So unfortunately, we'll have to keep it like this. If at some point there is a new documentation, a new API version, with a better documentation, we'll be able to adjust.